docs/metrics/rouge.md

```diff
@@ -0,0 +1,93 @@
+---
+search:
+  exclude: true
+---
+
+# ROUGE-N
+
+!!! info inline end "ROUGE vs. Recall"
+
+    Complimentary to BLEU, ROUGE-N can be thought of as an analog to [recall](recall.md) for text comparisons.
+
+ROUGE-N (**R**ecall-**O**riented **U**nderstudy for **G**isting **E**valuation), a metric within the broader ROUGE metric collection, is a vital metric in the field of natural language processing and text evaluation. It assesses the quality of a candidate text by measuring the overlap of n-grams between the candidate text and reference texts. ROUGE-N provides insights into the ability of a system to capture essential content and linguistic nuances, making it an important and versatile tool used in many NLP workflows. As the name implies, it is a recall-based metric - a complement to the precision-based BLEU score.
+
+## Implementation Details
+Formally, ROUGE-N is an n-gram recall between a candidate and a set of reference texts. That is, ROUGE-N calculates the number of overlapping n-grams between the generated and reference texts divided by the total number of n-grams in the reference texts.
+
+Mathematically, we define ROUGE-N as follows:
+
+$$
+\text{ROUGE-N} = \frac{\sum_{S \in \text{Reference Texts}} \sum_{\text{n-gram} \in S} \text{Match}(\text{n-gram})}{\sum_{S \in \text{Reference Texts}} \sum_{\text{n-gram} \in S} \text{Count}(\text{n-gram})}
```

Comment:
Just for my own curiosity, unlike BLEU score, ROUGE doesn't have a concept of brevity penalty.  How does it handle a large difference in the length of reference and candidate texts?

Reply:
Since it's a recall-based metric instead of precision like BLEU, the denominator (reference length) will be much larger than the numerator (matching n-grams) if the length of the candidate is too short. However, if the candidate is much longer than the reference, then it can still score perfect as long as it matches all n-grams in the reference (maximizing true positive count, like recall). In theory, BLEU and ROUGE are best used together since you don't want a candidate text that is massively long and contains all true positives but also a bunch of false positives.